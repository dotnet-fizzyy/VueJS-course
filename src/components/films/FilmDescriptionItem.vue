<template>
    <div :class="$style.root">
        <span :class="$style['key-label']">{{ keyLabel }}</span>

        <span v-if="!!additionalLabel" :class="$style['additional-label']">{{ additionalLabel }}</span>
    </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
    name: 'FilmDescriptionItem',
    props: {
        keyLabel: {
            type: String,
            required: true,
        },
        additionalLabel: {
            type: String,
        },
    },
});
</script>

<style lang="scss" module scoped>
@import '../../styles/fonts';

.root {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
}

.key-label {
    @extend .font;

    font-size: 20px;
    color: $dark-pink;
    margin-right: 5px;
}

.additional-label {
    @extend .font;

    font-size: 15px;
    color: $extra-light-grey;
}
</style>
