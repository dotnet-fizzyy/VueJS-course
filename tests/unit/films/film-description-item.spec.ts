import FilmDescriptionItem from '@/components/films/FilmDescriptionItem.vue';
import Vue from 'vue';
import { Wrapper, shallowMount } from '@vue/test-utils';

describe('FilmDescriptionItem.vue Tests', () => {
    const createComponent = (propsData): Wrapper<Vue> =>
        shallowMount(FilmDescriptionItem, {
            propsData,
        });

    it('Should render', () => {
        //Arrange
        const wrapper = createComponent({
            keyLabel: 'Key Label',
            additionalLabel: 'Addit Label',
        });

        //Act & Assert
        expect(wrapper.element).toMatchSnapshot();
    });
});
