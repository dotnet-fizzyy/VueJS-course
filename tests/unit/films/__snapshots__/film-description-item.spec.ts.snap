// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDescriptionItem.vue Tests Should render 1`] = `
<div
  class="root"
>
  <span
    class="key-label"
  >
    Key Label
  </span>
   
  <span
    class="additional-label"
  >
    Addit Label
  </span>
</div>
`;
